lib/src/block_parser.dart

```diff
@@ -54,10 +54,34 @@ class BlockParser {
   /// The markdown document this parser is parsing.
   final Document document;
 
+  /// The enabled block syntaxes. To turn a series of lines into blocks, each of
+  /// these will be tried in turn. Order matters here.
+  final List<BlockSyntax> blockSyntaxes = [];
+
   /// Index of the current line.
   int _pos;
 
-  BlockParser(this.lines, this.document) : _pos = 0;
+  /// The collection of built-in block parsers.
+  final List<BlockSyntax> standardBlockSyntaxes = const [
+    const EmptyBlockSyntax(),
+    const BlockHtmlSyntax(),
+    const SetextHeaderSyntax(),
+    const HeaderSyntax(),
+    const CodeBlockSyntax(),
+    const BlockquoteSyntax(),
+    const HorizontalRuleSyntax(),
+    const UnorderedListSyntax(),
+    const OrderedListSyntax(),
+    const ParagraphSyntax()
+  ];
+
+  BlockParser(this.lines, this.document) {
+    _pos = 0;
+    if (document.extensions.contains('fenced-code-blocks')) {
+      blockSyntaxes.add(const FencedCodeBlockSyntax());
+    }
+    blockSyntaxes.addAll(standardBlockSyntaxes);
```

Comment:
The way "extensions" or "plug-ins" work for InlineSyntaxes is that the parser pulls them from the document, which the public API can mutate. Outside code can just add in new objects that implement InlineSyntax.

I'm not sure if that's a great API or not—I lean towards not since it exposes an awful lot about how the parser is implemented—but I think it would be good for block syntax to be consistent with it.

You could either also add a `blockSyntaxes` field to Document, or remove `inlineSyntaxes` and use extensions for it.